scripts/Diffraction/single_crystal/base_sx.py

```diff
@@ -614,6 +618,180 @@ def remove_non_integrated_peaks(peaks):
+    ):
+        """
+        Bin MD workspace in peak region with projection axes given by the axes of the ellipsoid shape
+        :param wsMD: MD workspace (with 3 dims)
+        :param pk: Peak object that has been integrated
+        :param peak_shape: shape of integrated peak (spherical, ellipsoid or none)
+        :param nbins_max: number of bins along the major axis of the ellipsoid
+        :param extent: extent of output MD workspace along each dimension in units of the outer background radius
+        :param frame_to_peak_centre_attr: name of attribute to return peak centre in appropriate frame
+        :return ws_cut: MD workspace in peak region
+        :return radii: radii of the ellipsoid peak region
+        :return bg_inner_radii: inner background radii of the ellipsoid peak region
+        :return bg_outer_radii: outer background radii of the ellipsoid peak region
+        :return box_lengths: length of each dimension in the MD workspace
+        :return: imax: index of dimension with largest radius
+        """
+        ndims = wsMD.getNumDims()
+        radii, bg_inner_radii, bg_outer_radii, evecs, translation = BaseSX._get_ellipsoid_params_from_peak(peak_shape, ndims)
+        imax = np.argmax(radii)
+        # calc center in frame of ellipsoid axes
+        cen = getattr(pk, frame_to_peak_centre_attr)()
+        cen = np.matmul(evecs.T, np.array(cen) + translation)
+        # get extents
+        box_lengths = bg_outer_radii if not np.allclose(bg_outer_radii, 0.0) else radii
+        box_lengths = box_lengths * extent
+        extents = np.vstack((cen - box_lengths, cen + box_lengths))
+        # get nbins along each axis
+        nbins = np.array([int(nbins_max * radii[iax] / radii[imax]) for iax in range(ndims)])
+        # ensure minimum of 3 bins inside radius along each dim
+        min_nbins_in_radius = 3
+        nbins_in_radius = np.min(nbins * radii / box_lengths)  # along most coarsely binned dimension
+        if nbins_in_radius < min_nbins_in_radius:
+            nbins = nbins * min_nbins_in_radius / nbins_in_radius
+        # call BinMD
+        ws_cut = mantid.BinMD(
+            InputWorkspace=wsMD,
+            OutputWorkspace="__ws_cut",
+            AxisAligned=False,
+            BasisVector0=r"Q$_0$,unit," + ",".join(np.array2string(evecs[:, 0], precision=6).strip("[]").split()),
+            BasisVector1=r"Q$_1$,unit," + ",".join(np.array2string(evecs[:, 1], precision=6).strip("[]").split()),
+            BasisVector2=r"Q$_2$,unit," + ",".join(np.array2string(evecs[:, 2], precision=6).strip("[]").split()),
+            OutputExtents=extents.flatten(order="F"),
+            OutputBins=nbins.astype(int),
+            EnableLogging=False,
+            StoreInADS=False,
+        )
+        return ws_cut, radii, bg_inner_radii, bg_outer_radii, box_lengths, imax
+
+    @staticmethod
+    def _get_ellipsoid_params_from_peak(peak_shape: PeakShape, ndims: int):
```

Comment:
Sorry, my last comment about docstrings I promise! Can we add one for this function?